### ISSUE TITLE
`apm upgrade -c false` returns Request for package information failed: read ECONNRESET

### ISSUE THREAD

**Issue by blacktop** [atom/apm#519]
`apm upgrade -c false` returns Request for package information failed: read ECONNRESET

This started happening over the past few weeks.

I have nothing in my `~/.atom/.apmrc`

### OS
`OSX 10.11.3`

### apm
```bash
~ ❯❯❯ apm -v
apm  1.6.0
npm  2.13.3
node 0.10.40
python 2.7.11
git 2.7.2
```

**Comment by blacktop** [comment]
![screen shot 2016-03-04 at 9 31 10 am](https://cloud.githubusercontent.com/assets/6118188/13533045/e0ff3b66-e1eb-11e5-86fb-5cc8b561c617.png)

**Comment by lee-dohm** [comment]
This happens occasionally. Have you tried again?

**Comment by lee-dohm** [comment]
I mean, does this happen every time? Or does it only happen sometimes?

**Comment by blacktop** [comment]
It is happening every single time :(

**Comment by blacktop** [comment]
I have seen another issue where people were saying it was the version of requests or npm that apm was using?  Should I try to update one of those and how do I do that?

Thank you.

**Comment by lee-dohm** [comment]
Is it only update? Or installs too?

You can try updating if you think this is a proxy issue, though if you suspect that have you tried the instructions in the [README](https://github.com/atom/apm#using-a-proxy)?

(Sorry for the 20 questions, just trying to understand the shape of the problem.)

**Comment by blacktop** [comment]
@lee-dohm installs work fine.  No worries, I greatly appreciate the help.

No proxy, just at home using CentryLink.  I did, however, install bleeding edge node via `brew install node` but Atom has it's own version of node/npm so that should be affecting anything?

**Comment by blacktop** [comment]
`apm update --verbose`
https://gist.github.com/blacktop/432879f153aa3e7087e5

**Comment by blacktop** [comment]
I tried just updating individually it works 'most of the time'

https://gist.github.com/blacktop/91d8e57a834ea697bb35

So something about updating them all in one command fails every time, but one package at a time succeeds most of the time?

**Comment by Alhadis** [comment]
I've been getting this error 90% of the time for the last, uhm, month or so. Probably longer.

Doesn't matter how I try updating, I get a timeout whichever way:

<img src="https://cloud.githubusercontent.com/assets/2346707/15676397/5de0dc4c-2789-11e6-9120-1012c998f33f.png" width="382" alt="Figure 1" />

<img src="https://cloud.githubusercontent.com/assets/2346707/15676417/75b48bde-2789-11e6-92fa-3987f50a5739.png" width="600" alt="Figure 2" />

I thought Atom.io has been having prolonged server troubles lately, but judging from the lack of recent issues being reported, I'm wondering if the problem's really on my end.

Output of [`apm update --verbose`](http://pastebin.com/raw/iRsFzzr4).

**Comment by Alhadis** [comment]
Interesting:
```
Unable to download https://www.atom.io/api/packages/language-pascal/versions/0.18.1/tarball: getaddrinfo ENOTFOUND
```
I'm able to visit [https://atom.io](https://atom.io) okay, but [https://**www.**atom.io/](https://**www.**atom.io/) can't be reached:
<img width="754" src="https://cloud.githubusercontent.com/assets/2346707/15742487/9261c43e-2902-11e6-98de-446076010eab.png" alt="Figure 1" />
I managed to access `www.atom.io` version briefly just before (which redirected me to the www-less variant), but I'm wondering if these fluctuations in response could be connected. Have a look at [the feedback](http://pastebin.com/raw/EaEnnEe4): you'll notice APM has no issues connecting to `atom.io` to get metadata, but it hangs up when trying to reach `www.atom.io`. Incidentally, I was able to download the tarball from the www-less version okay (https://www.atom.io/api/packages/language-pascal/versions/0.18.1/tarball).

Assuming that `www` points to a different nameserver, there might be a regional issue happening here (in one of the successful installs, I saw `X-Served-By: cache-...-mel` in the response headers (which I'm guessing is short for "Melbourne", my city).

**Comment by blacktop** [comment]
@Alhadis yeah I believe this was on my end.  A combination of centuryLink + a new mesh router was somehow messing things up.  So connecting my computer directly to the centuryLink modem instead of via WIFI fixed it for some reason???

**Comment by as-cii** [comment]
@Alhadis: that's really interesting, after running `apm update --verbose` locally it doesn't seem like apm tries to reach a `www` url on my machine. Just to take apm out of the equation, could you check if you have the `ATOM_API_URL` or the `ATOM_PACKAGES_URL` environment variables set (or any other variable that's related to Atom)?

Thanks! 🙇

**Comment by Alhadis** [comment]
Nope, neither.

**Comment by blacktop** [comment]
Is there a way to make apm more tolerant of slow networks?
